Error: rpc error: code = Canceled desc = context canceled

### Terraform Version
Terraform v0.12.21
### Crash Output
```
datadog_logs_custom_pipeline.pipeline: Creating...

Error: rpc error: code = Canceled desc = context canceled


panic: interface conversion: interface {} is nil, not map[string]interface {}
2020-07-09T07:33:40.098Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: 
2020-07-09T07:33:40.099Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: goroutine 27 [running]:
2020-07-09T07:33:40.099Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogPipeline(0xc0006c6b60, 0x7, 0x0, 0xffffffffffffffff)
2020-07-09T07:33:40.099Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_logs_custom_pipeline.go:660 +0x40a
2020-07-09T07:33:40.099Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: github.com/terraform-providers/terraform-provider-datadog/datadog.resourceDatadogLogsPipelineCreate(0xc0006c6b60, 0x10aeae0, 0xc0000701c0, 0x2, 0x20673a0)
2020-07-09T07:33:40.100Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_logs_custom_pipeline.go:320 +0x84
2020-07-09T07:33:40.100Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc0001afc20, 0xc0007c5720, 0xc0007c7020, 0x10aeae0, 0xc0000701c0, 0x1168401, 0xc00098a298, 0xc000941da0)
2020-07-09T07:33:40.100Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-datadog/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:310 +0x363
2020-07-09T07:33:40.100Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000172000, 0xc0008ab9e8, 0xc0007c5720, 0xc0007c7020, 0xc00092e2c8, 0xc00091c3e0, 0x116a680)
2020-07-09T07:33:40.100Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-datadog/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/provider.go:294 +0x9c
2020-07-09T07:33:40.100Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00016c150, 0x1562a40, 0xc0006d3710, 0xc0001f7030, 0xc00016c150, 0xc0006d36b0, 0x118df40)
2020-07-09T07:33:40.100Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-datadog/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:885 +0x8a2
2020-07-09T07:33:40.100Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x12cecc0, 0xc00016c150, 0x1562a40, 0xc0006d3710, 0xc00040e300, 0x0, 0x0, 0x0, 0xc00057ca00, 0x655)
2020-07-09T07:33:40.100Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-datadog/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3305 +0x23e
2020-07-09T07:33:40.100Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0004f2480, 0x156aee0, 0xc000074d80, 0xc000030100, 0xc0001538f0, 0x2039440, 0x0, 0x0, 0x0)
2020-07-09T07:33:40.100Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-datadog/vendor/google.golang.org/grpc/server.go:1024 +0x4e5
2020-07-09T07:33:40.100Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: google.golang.org/grpc.(*Server).handleStream(0xc0004f2480, 0x156aee0, 0xc000074d80, 0xc000030100, 0x0)
2020-07-09T07:33:40.101Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-datadog/vendor/google.golang.org/grpc/server.go:1313 +0xe02
2020-07-09T07:33:40.101Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0000a4100, 0xc0004f2480, 0x156aee0, 0xc000074d80, 0xc000030100)
2020-07-09T07:33:40.101Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-datadog/vendor/google.golang.org/grpc/server.go:722 +0x9f
2020-07-09T07:33:40.101Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-07-09T07:33:40.101Z [DEBUG] plugin.terraform-provider-datadog_v2.11.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-datadog/vendor/google.golang.org/grpc/server.go:720 +0xa1
2020-07-09T07:33:40.103Z [DEBUG] plugin: plugin process exited: path=/tmp/datadog/terraform/logs/pipeline/.terraform/plugins/linux_amd64/terraform-provider-datadog_v2.11.0_x4 pid=182 error="exit status 2"
2020/07/09 07:33:40 [DEBUG] datadog_logs_custom_pipeline.pipeline: apply errored, but we're indicating that via the Error pointer rather than returning it: rpc error: code = Canceled desc = context canceled
2020/07/09 07:33:40 [TRACE] <root>: eval: *terraform.EvalMaybeTainted
2020/07/09 07:33:40 [TRACE] EvalMaybeTainted: datadog_logs_custom_pipeline.pipeline encountered an error during creation, so it is now marked as tainted
2020/07/09 07:33:40 [TRACE] <root>: eval: *terraform.EvalWriteState
2020/07/09 07:33:40 [TRACE] EvalWriteState: removing state object for datadog_logs_custom_pipeline.pipeline
2020/07/09 07:33:40 [TRACE] <root>: eval: *terraform.EvalApplyProvisioners
2020/07/09 07:33:40 [TRACE] EvalApplyProvisioners: datadog_logs_custom_pipeline.pipeline has no state, so skipping provisioners
2020/07/09 07:33:40 [TRACE] <root>: eval: *terraform.EvalMaybeTainted
2020/07/09 07:33:40 [TRACE] EvalMaybeTainted: datadog_logs_custom_pipeline.pipeline encountered an error during creation, so it is now marked as tainted
2020/07/09 07:33:40 [TRACE] <root>: eval: *terraform.EvalWriteState
2020/07/09 07:33:40 [TRACE] EvalWriteState: removing state object for datadog_logs_custom_pipeline.pipeline
2020/07/09 07:33:40 [TRACE] <root>: eval: *terraform.EvalIf
2020/07/09 07:33:40 [TRACE] <root>: eval: *terraform.EvalIf
2020/07/09 07:33:40 [TRACE] <root>: eval: *terraform.EvalWriteDiff
2020/07/09 07:33:40 [TRACE] <root>: eval: *terraform.EvalApplyPost
2020/07/09 07:33:40 [ERROR] <root>: eval: *terraform.EvalApplyPost, err: rpc error: code = Canceled desc = context canceled
2020/07/09 07:33:40 [ERROR] <root>: eval: *terraform.EvalSequence, err: rpc error: code = Canceled desc = context canceled
2020/07/09 07:33:40 [TRACE] [walkApply] Exiting eval tree: datadog_logs_custom_pipeline.pipeline
2020/07/09 07:33:40 [TRACE] vertex "datadog_logs_custom_pipeline.pipeline": visit complete
2020/07/09 07:33:40 [TRACE] dag/walk: upstream of "datadog_logs_pipeline_order.pipeline" errored, so skipping
2020/07/09 07:33:40 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2020/07/09 07:33:40 [TRACE] dag/walk: upstream of "provider.datadog (close)" errored, so skipping
2020/07/09 07:33:40 [TRACE] dag/walk: upstream of "root" errored, so skipping
2020-07-09T07:33:40.108Z [DEBUG] plugin: plugin exited



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

SECURITY WARNING: the "crash.log" file that was created may contain 
sensitive information that must be redacted before it is safe to share 
on the issue tracker.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

--- a/datadog/resource_datadog_logs_custom_pipeline.go
+++ b/datadog/resource_datadog_logs_custom_pipeline.go
@@ -6,6 +6,7 @@ import (
 
 	datadogV1 "github.com/DataDog/datadog-api-client-go/api/v1/datadog"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
 )
 
 const (
@@ -126,11 +127,8 @@ var categoryProcessor = &schema.Schema{
 					"filter": {
 						Type:     schema.TypeList,
 						Required: true,
-						Elem: &schema.Resource{
-							Schema: map[string]*schema.Schema{
-								"query": {Type: schema.TypeString, Required: true},
-							},
-						},
+						MaxItems: 1,
+						Elem:     getFilterSchema(),
 					},
 					"name": {Type: schema.TypeString, Required: true},
 				},
@@ -657,7 +655,11 @@ func buildDatadogPipeline(d *schema.ResourceData) (*datadogV1.LogsPipeline, erro
 	ddPipeline.SetName(d.Get("name").(string))
 	ddPipeline.SetIsEnabled(d.Get("is_enabled").(bool))
 	if tfFilter := d.Get("filter").([]interface{}); len(tfFilter) > 0 {
-		ddPipeline.SetFilter(buildDatadogFilter(tfFilter[0].(map[string]interface{})))
+		filter, ok := tfFilter[0].(map[string]interface{})
+		if !ok {
+			return nil, fmt.Errorf("Couldn't build filter, is query empty?")
+		}
+		ddPipeline.SetFilter(buildDatadogFilter(filter))
 	}
 	ddProcessors, err := buildDatadogProcessors(d.Get("processor").([]interface{}))
 	if err != nil {
@@ -1068,11 +1070,7 @@ func getPipelineSchema(isNested bool) map[string]*schema.Schema {
 		"filter": {
 			Type:     schema.TypeList,
 			Required: true,
-			Elem: &schema.Resource{
-				Schema: map[string]*schema.Schema{
-					"query": {Type: schema.TypeString, Required: true},
-				},
-			},
+			Elem:     getFilterSchema(),
 		},
 		"processor": {
 			Type:     schema.TypeList,
@@ -1100,4 +1098,16 @@ func getProcessorSchema(isNested bool) map[string]*schema.Schema {
 		processorsSchema[tfProcessorType] = processor
 	}
 	return processorsSchema
+}
+
+func getFilterSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"query": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+	}
 }

